jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/editor/CloudWatchLogStream.kt

```diff
@@ -80,85 +79,53 @@ class CloudWatchLogStream(
         logsTable.autoResizeMode = JTable.AUTO_RESIZE_ALL_COLUMNS
         logsPanel.verticalScrollBar.addAdjustmentListener {
             if (logsTable.model.rowCount == 0) {
                 return@addAdjustmentListener
             }
             if (logsPanel.verticalScrollBar.isAtBottom()) {
-                launch {
-                    runUnlessDisposed(this@CloudWatchLogStream) {
-                        val items = logStreamClient.loadMoreForward()
-                        if (items.isNotEmpty()) {
-                            val events = logsTable.logsModel.items.plus(items)
-                            withContext(edtContext) { logsTable.logsModel.items = events }
-                        }
-                    }
-                }
+                launch { logStreamClient.channel.send(CloudWatchLogStreamCoroutine.Messages.LOAD_FORWARD) }
             } else if (logsPanel.verticalScrollBar.isAtTop()) {
-                launch {
-                    runUnlessDisposed(this@CloudWatchLogStream) {
-                        val items = logStreamClient.loadMoreBackward()
-                        if (items.isNotEmpty()) {
-                            val events = items.plus(logsTable.logsModel.items)
-                            withContext(edtContext) { logsTable.logsModel.items = events }
-                        }
-                    }
-                }
+                launch { logStreamClient.channel.send(CloudWatchLogStreamCoroutine.Messages.LOAD_BACKWARD) }
             }
         }
         launch {
-            runUnlessDisposed(this@CloudWatchLogStream) {
-                try {
-                    val items = if (startTime != null && duration != null) {
-                        logStreamClient.loadInitialAround(startTime, duration)
-                    } else {
-                        logStreamClient.loadInitial(fromHead)
-                    }
-                    withContext(edtContext) {
-                        logsTable.emptyText.text = message("cloudwatch.logs.no_events")
-                        logsTable.logsModel.items = items
-                        logsTable.setPaintBusy(false)
-                    }
-                } catch (e: Exception) {
-                    val errorMessage = message("cloudwatch.logs.failed_to_load_stream", logStream)
-                    CloudWatchLogGroup.LOG.error(e) { errorMessage }
-                    notifyError(title = errorMessage, project = project)
-                    withContext(edtContext) { logsTable.emptyText.text = errorMessage }
+            try {
+                if (startTime != null && duration != null) {
+                    logStreamClient.loadInitialAround(startTime, duration)
+                } else {
+                    logStreamClient.loadInitial()
                 }
+            } catch (e: Exception) {
+                val errorMessage = message("cloudwatch.logs.failed_to_load_stream", logStream)
+                CloudWatchLogGroup.LOG.error(e) { errorMessage }
```

Comment:
Wrong LOG?, also why is that allowed, did we make the LOG public?

Reply:
OK that's weird, fixed and made that LOG private